v2/bench_test.go

```diff
@@ -47,3 +55,93 @@ func BenchmarkReadBlocks(b *testing.B) {
 		}
 	})
 }
+
+// BenchmarkExtractV1File extracts inner CARv1 payload from a sample CARv2 file using ExtractV1File.
+func BenchmarkExtractV1File(b *testing.B) {
+	path := filepath.Join(b.TempDir(), "bench-large-v2.car")
+	generateRandomCarV2File(b, path, 10*1024*1024) // 10 MiB
+	defer os.Remove(path)
+
+	info, err := os.Stat(path)
+	if err != nil {
+		b.Fatal(err)
+	}
+	b.SetBytes(info.Size())
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	b.RunParallel(func(pb *testing.PB) {
+		dstPath := filepath.Join(b.TempDir(), "destination.car")
+		for pb.Next() {
+			err = carv2.ExtractV1File(path, dstPath)
+			if err != nil {
+				b.Fatal(err)
+			}
+			_ = os.Remove(dstPath)
+		}
+	})
+}
+
+// BenchmarkExtractV1UsingReader extracts inner CARv1 payload from a sample CARv2 file using Reader
+// API. This benchmark is implemented to be used as a comparison in conjunction with
+// BenchmarkExtractV1File.
+func BenchmarkExtractV1UsingReader(b *testing.B) {
+	path := filepath.Join(b.TempDir(), "bench-large-v2.car")
+	generateRandomCarV2File(b, path, 10*1024*1024) // 10 MiB
+	defer os.Remove(path)
+
+	info, err := os.Stat(path)
+	if err != nil {
+		b.Fatal(err)
+	}
+	b.SetBytes(info.Size())
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	b.RunParallel(func(pb *testing.PB) {
+		dstPath := filepath.Join(b.TempDir(), "destination.car")
+		for pb.Next() {
+			dst, err := os.Create(dstPath)
+			if err != nil {
+				b.Fatal(err)
+			}
+			reader, err := carv2.OpenReader(path)
+			if err != nil {
+				b.Fatal(err)
+			}
+			_, err = io.Copy(dst, reader.DataReader())
+			if err != nil {
+				b.Fatal(err)
+			}
+			if err := dst.Close(); err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
+}
+
+func generateRandomCarV2File(b *testing.B, path string, minTotalBlockSize int) {
+	bs, err := blockstore.OpenReadWrite(path, []cid.Cid{})
+	defer func() {
+		if err := bs.Finalize(); err != nil {
+			b.Fatal(err)
+		}
+	}()
+	if err != nil {
+		b.Fatal(err)
+	}
+	buf := make([]byte, 1024)
```

Comment:
nit: I think it would be slightly more realistic if the blocks are a bit larger, like 32KiB. The block sizes could really be anything, but in general one can guess that the average block size will be in the tens-to-hundreds of kilobytes.

for example, I believe IPFS starts chunking files into blocks if they are over 256KiB.